### ISSUE TITLE
False positive on conversion operators, that take arguments by reference

### ISSUE THREAD

**Issue by lostmsu** [ufcpp/NonCopyableAnalyzer#14]
```csharp
[NonCopyable]
ref struct NewReference
{
  // this should be allowed to be called any time,
  // because it takes input by reference (e.g. `in` is used)
  public static implicit operator BorrowedReference(in NewReference reference)
    => new BorrowedReference(reference.pointer);
}
...
NewReference @new = new NewReference();
BorrowedReference borrowed = @new; // "error NoCopy06: 🚫 conversion"
```

**Comment by lostmsu** [comment]
BTW, cool project! Started using it in [Python.NET](https://github.com/pythonnet/pythonnet). Rust-style lifetime tracking FTW!

Meta BTW: cool, GitHub shows us as a user in "UsedBy".

**Comment by ufcpp** [comment]
Thank you so much for your great feedback! I'm honored to hear that😊 However, I'm too busy to re-build this project now. It's helpful for me if you send me pull request and I will accept it! Thanks again your feedback!!

**Comment by ufcpp** [comment]
notes:

- Not supported yet: compound assignment, unary operator, binary operator
  -  https://github.com/ufcpp/NonCopyableAnalyzer/blob/master/src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs#L169
- Read-only function member invocation could be allowed
  - https://github.com/ufcpp/NonCopyableAnalyzer/blob/master/src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs#L133
  - https://github.com/ufcpp/NonCopyableAnalyzer/blob/master/src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs#L141
